cmake/modules/extensions.cmake

```diff
@@ -5874,9 +5874,31 @@ function(add_llext_target target_name)
   set(llext_lib_target ${target_name}_llext_lib)
   if(CONFIG_LLEXT_TYPE_ELF_OBJECT)
 
-    # Create an object library to compile the source file
-    add_library(${llext_lib_target} EXCLUDE_FROM_ALL OBJECT ${source_files})
-    set(llext_lib_output $<TARGET_OBJECTS:${llext_lib_target}>)
+    get_property(TOPT GLOBAL PROPERTY TOPT)
+    get_property(COMPILER_TOPT TARGET compiler PROPERTY linker_script)
+    set_ifndef(  TOPT "${COMPILER_TOPT}")
+    set_ifndef(  TOPT -Wl,-T)  # Use this if the compiler driver doesn't set a value
+
+    set(no_gc_sections_flag)
+    if("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT})
```

Comment:
we have the toolchain infrastructure in:
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/toolchain
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/compiler
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/linker

where the toolchain / compiler / linker can describe flags as properties and then those can be fetched using property name instead.

Please rework.